Shopify/dawn
# Fix double margins in RTE when non-paragraphs are used

### PR Summary: 

Fixes unintentional double margins generated by non-paragraph tags in the Rich Text editor. 

### Why are these changes introduced?

Fixes #2416

Recent updates to the Rich Text editor now allow it to use headings and lists, in addition to paragraphs. Dawn today clears out the top and bottom margins of paragraphs inside of the Rich Text editor. These new tags were not targeted by our existing rule, resulting in more vertical space between elements than we'd originally designed for there to be. 

### What approach did you take?

I used a wildcard here — I can't think of a reason why we _wouldn't_ want to clear these extra margins. But if folks feel otherwise, it's easy enough to just add a bunch of rules for all the currently-supported tags instead (`.rte > h1:first-child`, `.rte > h2:first-child`, etc.). The downside is just that it's far more code, and could break again if we allow more tags in here in the future. 

### Visual impact on existing themes

Paragraph Tags (No Change)

Before|After
---|---
<img width="920" alt="both-p" src="https://user-images.githubusercontent.com/1202812/225339936-c3d6d763-4739-45b7-b9c0-ca0ddf0580ec.png">|<img width="920" alt="both-p" src="https://user-images.githubusercontent.com/1202812/225339936-c3d6d763-4739-45b7-b9c0-ca0ddf0580ec.png">

H2 Tags

Before|After
---|---
<img width="921" alt="before-h2" src="https://user-images.githubusercontent.com/1202812/225340037-a30b38a6-d150-4249-8805-794ce40ab28c.png">|<img width="920" alt="after-h2" src="https://user-images.githubusercontent.com/1202812/225340032-b9bac2f1-a619-4f4f-af7f-05e3453be065.png">

H3 Tags
Before|After
---|---
<img width="920" alt="before-h3" src="https://user-images.githubusercontent.com/1202812/225340127-f1dd7353-21ba-4d04-990d-a7f440123470.png">|<img width="919" alt="after-h3" src="https://user-images.githubusercontent.com/1202812/225340123-42fa0c2c-33eb-4270-8608-cb027e35d4c5.png">

This also effects `ul`, `ol`, and `h1-h6`, but the effect is the same. 


### Testing steps/scenarios

1. Open the demo store. 
2. Add an image with Text Block
3. Edit the description text style so that it uses headings or lists. 
4. Note that the vertical margins between the heading + description + button stay consistent. 
5. Check other uses of `rte` and look for visual changes. (i.e. product descriptions, blog posts, and pages). There should be no visual change when this PR is active. 

### Demo links

- [Store](https://vzlclinzni9yi7s0-45840990230.shopifypreview.com)
- [Editor](https://os2-demo.myshopify.com/admin/themes/139533451286/editor)

### Checklist
